demo/dotnet/PvSpeakerDemo/PvSpeakerDemo.cs

```diff
@@ -0,0 +1,246 @@
+                    throw new InvalidDataException("Not a valid WAV file.");
+
+                binaryReader.BaseStream.Seek(4, SeekOrigin.Current);
+                string format = new string(binaryReader.ReadChars(4));
+                if (format != "WAVE")
+                    throw new InvalidDataException("Not a valid WAV file.");
+
+                string formatChunkId = new string(binaryReader.ReadChars(4));
+                if (formatChunkId != "fmt ")
+                    throw new InvalidDataException("Invalid WAV file format.");
+
+                int formatChunkSize = binaryReader.ReadInt32();
+                int audioFormat = binaryReader.ReadInt16();
+                int channels = binaryReader.ReadInt16();
+                int sampleRate = binaryReader.ReadInt32();
+                int byteRate = binaryReader.ReadInt32();
+                int blockAlign = binaryReader.ReadInt16();
+                int bitsPerSample = binaryReader.ReadInt16();
+
+                if (formatChunkSize > 16)
+                    binaryReader.BaseStream.Seek(formatChunkSize - 16, SeekOrigin.Current);
+
+                if (channels != 1)
+                    throw new InvalidDataException("WAV file must have a single channel (MONO)");
+
+                string dataChunkId = new string(binaryReader.ReadChars(4));
+                if (dataChunkId != "data")
+                    throw new InvalidDataException("Invalid WAV file data chunk.");
+
+                int dataChunkSize = binaryReader.ReadInt32();
+                byte[] audioData = binaryReader.ReadBytes(dataChunkSize);
+
+                return new WavFileInfo
+                {
+                    SampleRate = sampleRate,
+                    BitsPerSample = bitsPerSample,
+                    AudioData = audioData
+                };
+            }
+        }
+
+        /// <summary>
+        /// Lists available audio input devices.
+        /// </summary>
+        public static void ShowAudioDevices()
+        {
+            string[] devices = PvSpeaker.GetAvailableDevices();
+            for (int i = 0; i < devices.Length; i++)
+            {
+                Console.WriteLine($"index: {i}, device name: {devices[i]}");
+            }
+        }
+
+        private static readonly string HELP_STR = "Available options: \n" +
+            " --show_audio_devices: List of audio devices currently available for use.\n" +
+            " --audio_device_index: Index of output audio device.\n" +
+            " --input_wav_path: Path to PCM WAV file to be played.\n" +
+            " --buffer_size_secs: Size of internal PCM buffer in seconds.\n" +
+            " --output_wav_path: Path to the output WAV file where the PCM data passed to PvSpeaker will be written.\n";
+
+        static void Main(string[] args)
+        {
+            if (args.Length == 0)
+            {
+                Console.WriteLine(HELP_STR);
+                Console.Read();
+                return;
+            }
```

Comment:
I think we should also allow them to type `--help` or `-h` for usage. We do this in our other .NET demos I'm pretty sure